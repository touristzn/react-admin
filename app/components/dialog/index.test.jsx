import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import DiaLog from './index'

let wrapper

describe('<DiaLog />', () => {
  const state = {
    // 适用场景：根据请求数据有值或无值，设置按钮能否点击
    isDisabled: false,
  }

  const props = {
    title: 'dialog',
    onClose: jest.fn(),
    buttons: [
      {
        text: 'cancel',
        onClick: jest.fn()
      },
      {
        text: 'save',
        type: state.isDisabled ? 'primary' : 'disabled',
        onClick: jest.fn()
      },
    ],
  }

  beforeEach(() => {
    wrapper = shallow(
      <DiaLog {...props}>内容...</DiaLog>
    )
  })

  it('render header', () => {
    expect(wrapper.find('h1').text()).toBe(props.title)
    wrapper.find('header button').simulate('click')
    expect(props.onClose).toHaveBeenCalled()
  })

  it('render the footer buttons', () => {
    expect(wrapper.find('footer Button')).toHaveLength(2)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})