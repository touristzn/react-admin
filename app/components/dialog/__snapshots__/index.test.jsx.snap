// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DiaLog /> renders correctly 1`] = `
<div
  className="popup-box"
>
  <div
    className="dialog-win"
  >
    <header>
      <h1>
        dialog
      </h1>
      <button
        onClick={
          [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          }
        }
      >
        Close
      </button>
    </header>
    <div
      className="dialog-win_content"
    >
      内容...
    </div>
    <footer>
      <Button
        key="0"
        label="cancel"
        onClick={[MockFunction]}
      />
      <Button
        key="1"
        label="save"
        onClick={[MockFunction]}
        type="disabled"
      />
    </footer>
  </div>
</div>
`;
